src/map/script.cpp

```diff
@@ -25686,6 +25722,221 @@ BUILDIN_FUNC( openstylist ){
 #endif
 }
 
+BUILDIN_FUNC(getitemdata)
+{
+	struct map_session_data* sd;
+
+	if (!script_rid2sd(sd)) {
+		st->state = END;
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	struct script_data *data = script_getdata(st, 3);
+
+	if (!data_isreference(data)) {
+		ShowWarning("buildin_getitemdata: Error in argument! Please give a variable to store values in.\n");
+		st->state = END;
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	int idx = script_getnum(st, 2);
+	if (idx < 0 || idx >= MAX_INVENTORY) {
+		ShowWarning("buildin_getitemdata: Index %d is out of the range 0-%d.\n", idx, MAX_INVENTORY - 1);
+		st->state = END;
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	if (sd->inventory_data[idx] == nullptr) {
+		ShowWarning("buildin_getitemdata: No item can be deleted from index %d of player %s (AID: %u, CID: %u).\n", idx, sd->status.name, sd->status.account_id, sd->status.char_id);
+		st->state = END;
+		return SCRIPT_CMD_FAILURE;
+	}
+	if (sd->inventory_data[idx]->type != IT_ARMOR && sd->inventory_data[idx]->type != IT_WEAPON && sd->inventory_data[idx]->type != IT_SHADOWGEAR) {
```

Comment:
if !itemdb_isequip2?

Reply:
Maybe I could just remove that check for `getitemdata`